osu.Game/Overlays/UserProfileOverlay.cs

```diff
@@ -149,30 +151,32 @@ private void userLoadComplete(User user)
             }
         }
 
-        private class ProfileTabControl : OverlayTabControl<ProfileSection>
+        private class ProfileSectionTabControl : OverlayTabControl<ProfileSection>
         {
-            public ProfileTabControl()
+            public ProfileSectionTabControl()
             {
                 TabContainer.RelativeSizeAxes &= ~Axes.X;
                 TabContainer.AutoSizeAxes |= Axes.X;
                 TabContainer.Anchor |= Anchor.x1;
                 TabContainer.Origin |= Anchor.x1;
             }
 
-            protected override TabItem<ProfileSection> CreateTabItem(ProfileSection value) => new ProfileTabItem(value)
+            protected override TabItem<ProfileSection> CreateTabItem(ProfileSection value) => new ProfileSectionTabItem(value)
             {
-                AccentColour = AccentColour
+                // This needs to be set here manually because TabControl attempts to propagate these values to its children before these tabs are added
+                AccentColour = AccentColour,
+                TextColour = TextColour
```

Comment:
Can you explain further? I'm kind of confused at why this is necessary.

Reply:
At the moment these tabs are created, its parent `TabControl` has already attempted to set its children `AccentColour` / `TextColour` like here:
https://github.com/ppy/osu/blob/6b284fb3e74fbfce50c96e1096563360b95ba9cd/osu.Game/Graphics/UserInterface/OsuTabControl.cs#L29-L37

 This happens because they're being set when the `TabControl` is created, while the tabs in `ProfileSectionTabControl` are added a bit later (when their data is loaded up from the API):

https://github.com/ppy/osu/blob/6b284fb3e74fbfce50c96e1096563360b95ba9cd/osu.Game/Overlays/UserProfileOverlay.cs#L139-L149

`AccentColour` was already being set this way here, so I just added the `TextColour` which acts pretty much in the same manner.

Reply:
That is immensely broken & yet another argument to just move `OverlayColourProvider` down to the tabs themselves.

Reply:
I moved the text colour change down to the tab items but left `AccentColour` as is, since that's being set by the `OsuTabControl` higher up in the hierarchy and I don't think I want to change that.

Reply:
A half-change feels worse than the existing state of affairs to be honest. Let's bring this back as it was.